crates/bevy_winit/src/lib.rs

```diff
@@ -192,6 +193,64 @@ pub fn winit_runner(mut app: App) {
                         });
                     }
                 },
+                WindowEvent::Touch(
+                    touch
+                    @
+                    Touch {
+                        phase: TouchPhase::Started,
+                        ..
+                    },
+                ) => {
+                    let mut touch_finger_input_events =
+                        app.resources.get_mut::<Events<TouchFingerInput>>().unwrap();
+                    touch_finger_input_events.send(converters::convert_touch_input(
+                        ElementState::Pressed,
+                        &touch,
+                    ));
+                }
+                WindowEvent::Touch(
+                    touch
+                    @
+                    Touch {
+                        phase: TouchPhase::Ended,
+                        ..
+                    },
+                ) => {
+                    let mut touch_finger_input_events =
+                        app.resources.get_mut::<Events<TouchFingerInput>>().unwrap();
+                    touch_finger_input_events.send(converters::convert_touch_input(
+                        ElementState::Released,
+                        &touch,
+                    ));
+                }
+                WindowEvent::Touch(
+                    touch
+                    @
+                    Touch {
+                        phase: TouchPhase::Cancelled,
+                        ..
+                    },
+                ) => {
+                    let mut touch_finger_input_events =
+                        app.resources.get_mut::<Events<TouchFingerInput>>().unwrap();
+                    touch_finger_input_events.send(converters::convert_touch_input(
+                        ElementState::Released,
+                        &touch,
```

Comment:
I'm not sure we should be mapping `Cancelled` to `Released`. The `Input` abstraction as it stands doesn't map well to touch. I think either we extend `Input` or create a different abstraction for touch... I think I personally would prefer the latter.

Reply:
I agree with this. We should mirror the winit TouchPhase enum in the bevy events.